src/TLuaInterpreter.cpp

```diff
@@ -15289,7 +15288,7 @@ void TLuaInterpreter::msdp2Lua(const char* src)
             if (last == MSDP_VAL || last == MSDP_TABLE_CLOSE || last == MSDP_ARRAY_CLOSE) {
                 script.append(',');
             }
-            if (transcodedSrc.at(i + 1) != MSDP_TABLE_OPEN && transcodedSrc.at(i + 1) != MSDP_ARRAY_OPEN) {
+            if ((transcodedSrc.length() > i + 1) && transcodedSrc.at(i + 1) && transcodedSrc.at(i + 1) != MSDP_TABLE_OPEN && transcodedSrc.at(i + 1) != MSDP_ARRAY_OPEN) {
```

Comment:
Um, you already have the value on the left of the first condition as `textLength` - no need to call `QByteArray::length()` again!

Reply:
Totally do. Thanks for the catch!